packages/components/src/Combobox/components/ComboboxContent/ComboboxContent.tsx

```diff
@@ -115,8 +117,24 @@ export function ComboboxContent(props: ComboboxContentProps): JSX.Element {
   return ReactDOM.createPortal(template, document.body);
 
   function handleSelection(selection: ComboboxOption) {
-    props.onSelect(selection);
-    setSearchValue("");
-    setOpen(false);
+    if (multiselect) {
+      handleMultiSelect(props.onSelect, props.selected, selection);
+    } else {
+      props.onSelect([selection]);
+      setSearchValue("");
+      setOpen(false);
+    }
+  }
+}
+
+function handleMultiSelect(
+  selectCallback: (selected: ComboboxOption[]) => void,
+  selected: ComboboxOption[],
+  selection: ComboboxOption,
+) {
+  if (selected.some(s => s.id === selection.id)) {
+    selectCallback(selected.filter(s => s.id !== selection.id));
```

Comment:
thinking about how we might optimize this

`some` is fine because we do need to find if it was already selected, and it short circuits once it finds a result

the `filter` though. is there a smarter way to do that that doesn't involve iterating over all of the items? 🤔

Reply:
kk so I'm thinking about this and I wouldn't say this code is giving me any super big red flags in terms of efficiency. Mainly because we aren't dealing with any extremely large data sets here typically so we should be ok with this. Like one thing I was thinking is that if we really did want to optimize this we could always maintain a separate data structure using `Set` to store the object references to keep track of selected items by their IDs for faster lookups. then we could use the `has` method to check if the ID of the option is present? Let me know if that makes sense but I also think that might just be overkill considering what I mentioned initially.

Reply:
There's definitely room for improvement if we need it, but I side with Diane a bit here in that it could be  premature optimization that may never be needed. What we could do though, is mark this as a 'known area for improvement' so that if we do start to see complaints of combobox being slow, we'll know where to start looking.

Reply:
I agree with both of you, that while there are things we _could_ do here to squeeze as much optimization as possible out of this, we don't currently have any performance problems to speak of

I'm fine leaving it as is, and if someday we start seeing problems around this, we've got some ideas ready to go